### PR TITLE
Adding Agent Pool associations to OAuth Client for Private VCS

### PR DESCRIPTION
**Pull request by zainq11**
### What

Private VCS allows us to associate an Agent Pool to an OAuth Client. This change documents the  API changes to support that.

### Why

Users can now configure Private VCS connections via the API. The API also serves as the backend for `go-tfe` and `terraform-provider-tfe`. The provider [PR](https://github.com/hashicorp/terraform-provider-tfe/pull/1255) is already merged.

### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [X] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [X] Description links to related pull requests or issues, if any.

#### Content
- [ ] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [ ] API documentation and the API Changelog have been updated. 
- [ ] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [ ] Pages with related content are updated and link to this content when appropriate.
- [ ] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [ ] New pages have metadata (page name and description) at the top.
- [ ] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [ ] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 1]
```diff
@@ -185,6 +193,14 @@ curl \
         "links": {
           "related": "/api/v2/oauth-tokens/ot-KaeqH4cy72VPXFQT"
         }
+      },
+      "agent-pool": {
+        "data": { 
+          "id": "apool-VsmjEMcYkShrckpK", "type": "agent-pools"
```
Comment (aaabdelgany) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:199]: nit: newline for type

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 2]
```diff
@@ -254,6 +271,10 @@ Properties without a default value are required.
         "data": [
           { "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }
         ]
+      },
+      "agent-pool": {
+        "data": 
+          { "id": "apool-VsmjEMcYkShrckzzz", "type": "agent-pools" }
```
Comment (aaabdelgany) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:277]: same nit as above

File: website/docs/cloud-docs/api-docs/oauth-clients.mdx [hunk 3]
```diff
@@ -235,6 +251,7 @@ Properties without a default value are required.
 | `data.attributes.rsa-public-key`     | string |         | **Required for Bitbucket Data Center in conjunction with the `secret`. Not used for any other providers.** The text of the SSH public key associated with your Bitbucket Data Center application link.   |
 | `data.attributes.organization-scoped` | boolean |         | Whether or not the OAuth client is scoped to all projects and workspaces in the organization. Defaults to `true`.
 | `data.relationships.projects.data[]` | array\[object] | `[]`      | A list of resource identifier objects that defines which projects are associated with the OAuth client. If `data.attributes.organization-scoped` is set to `false`, the OAuth client is only available to this list of projects. Each object in this list must contain a project `id` and use the `"projects"` type. For example, `{ "id": "prj-AwfuCJTkdai4xj9w", "type": "projects" }`. |
+| `data.relationships.agent-pool.data` | object | `{}`    | **For Private VCS** The Agent Pool associated to the VCS connection. This pool will be responsible for forwarding VCS Provider API calls and cloning VCS repositories.                         |
```
Comment (aaabdelgany) [inline website/docs/cloud-docs/api-docs/oauth-clients.mdx:254]: do we have any other pages that we can link to that explain what private vcs is?

Comment (paulanunda) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:255]: Do we _need_ to specify if an attribute is only for Private VCS?

Comment (roleesinhaHC) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx:255]: > do we have any other pages that we can link to that explain what private vcs is?

AFAIK the only public doc that mentions private-vcs is the [feature entitlements one](https://developer.hashicorp.com/terraform/enterprise/api-docs).

Comment (zainq11) [reply website/docs/cloud-docs/api-docs/oauth-clients.mdx]: Thanks everyone. I went ahead and removed the reference to Private VCS. We can reference this setting in the Private VCS docs instead and keep this page agnostic.

**Review by paulanunda — APPROVED** [review]

**Review by aaabdelgany — APPROVED** [review]

**Pull request merged**